app/models/host.rb

```diff
@@ -41,5 +39,21 @@ def filter_by_compliance_score(_filter, operator, score)
 
       { conditions: "hosts.id IN(#{ids.join(',')})" }
     end
+
+    def has_test_results(_filter, operator, value)
+      ids = Host.select do |h|
+        h.test_results.present? == ActiveModel::Type::Boolean.new.cast(value)
+      end
+      ids = ids.pluck(:id).map { |id| "'#{id}'" }
+      scoped_search_sql_conditions(ids, operator)
+    end
+
+    def scoped_search_sql_conditions(ids, operator)
+      return { conditions: '1=0' } if ids.empty?
+      operator = operator == '<>' ? 'NOT' : ''
```

Comment:
`<>` is not a valid operator. scoped_search really ought to have a query validator :/

```suggestion
      operator = operator == '!=' ? 'NOT' : ''
```

Reply:
https://github.com/RedHatInsights/compliance-backend/pull/365#discussion_r398627244